SimPEG/directives/directives.py

```diff
@@ -2205,66 +2205,224 @@ def JtJv(v):
+
+    Notes
+    -----
+    Let :math:`\mathbf{J}` represent the Jacobian. To create sensitivity weights, root-mean squared (RMS) sensitivities
+    :math:`\mathbf{s}` are computed by summing the squares of the rows of the Jacobian:
+
+    .. math::
+        \mathbf{s} = \Bigg [ \sum_i \, \mathbf{J_{i, \centerdot }}^2 \, \Bigg ]^{1/2}
+
+    The dynamic range of RMS sensitivities can span many orders of magnitude. When computing sensitivity
+    weights, thresholding is generally applied to set a minimum value.
+
+    Thresholding
+    ^^^^^^^^^^^^
+
+    If **global** thresholding is applied, we add a constant :math:`\tau` to the RMS sensitivities:
+
+    .. math::
+        \mathbf{\tilde{s}} = \mathbf{s} + \tau
+
+    In the case of **percentile** thresholding, we let :math:`s_{\%}` represent a given percentile.
+    Thresholding to set a minimum value is applied as follows:
+
+    .. math::
+        \tilde{s}_j = \begin{bases}
+        s_j \;\; for \;\; s_j \geq s_{\%} \\
+        s_{\%} \;\; for \;\; s_j < s_{\%}
+        \end{cases}
+
+    If **absolute** thresholding is applied, we define :math:`\eta` as a fractional percent.
+    In this case, thresholding is applied as follows:
+
+    .. math::
+        \tilde{s}_j = \begin{bases}
+        s_j \;\; for \;\; s_j \geq \eta s_{max} \\
+        \eta s_{max} \;\; for \;\; s_j < \eta s_{max}
+        \end{cases}
     """
-    Directive to take care of re-weighting
-    the non-linear problems. Assumes that the map of the regularization
-    function is either Wires or Identity.
-    Good for any problem where J is formed explicitly.
-    """
 
-    def __init__(self, everyIter=True, threshold=1e-12, normalization=True, **kwargs):
+    def __init__(
+        self,
+        every_iteration=False,
+        threshold_value=1e-12,
+        threshold_method="global",
+        normalization_method="maximum",
+        **kwargs,
+    ):
+
+        if "everyIter" in kwargs.keys():
+            warnings.warn(
+                "'everyIter' property is deprecated and will be removed in SimPEG 0.19.0."
+                "Please use 'every_iteration'."
+            )
+            every_iteration = kwargs.pop("everyIter")
+
+        if "threshold" in kwargs.keys():
+            warnings.warn(
+                "'threshold' property is deprecated and will be removed in SimPEG 0.19.0."
+                "Please use 'threshold_value'."
+            )
+            threshold_value = kwargs.pop("threshold")
+
+        if "normalization" in kwargs.keys():
+            warnings.warn(
+                "'normalization' property is deprecated and will be removed in SimPEG 0.19.0."
+                "Please define normalization using 'normalization_method'."
+            )
+            normalization_method = kwargs.pop("normalization")
+            if normalization_method is True:
```

Comment:
These lines are repeated in the `setter` of `normalization_method` you can just let that function handle changing it from a `bool` to the appropriate `str` (or None) option. Then you also get a message to the user about how boolean input is also deprecated.